source/frontend/src/features/leases/detail/LeasePages/deposits/components/DepositsReceivedContainer/columns.tsx

```diff
@@ -11,44 +11,47 @@ import TooltipIcon from '@/components/common/TooltipIcon';
 import { ColumnWithProps, renderDate, renderMoney } from '@/components/Table';
 import Claims from '@/constants/claims';
 import useKeycloakWrapper from '@/hooks/useKeycloakWrapper';
-import { Api_Contact } from '@/models/api/Contact';
-import { Api_SecurityDeposit } from '@/models/api/SecurityDeposit';
+import { ApiGen_Concepts_Contact } from '@/models/api/generated/ApiGen_Concepts_Contact';
+import { ApiGen_Concepts_SecurityDeposit } from '@/models/api/generated/ApiGen_Concepts_SecurityDeposit';
 import { formatNames } from '@/utils/personUtils';
+import { exists } from '@/utils/utils';
 
 export class DepositListEntry {
   public id: number;
   public depositTypeDescription: string;
   public depositDescription: string;
   public amountPaid: number;
   public paidDate: string;
-  public contactHolder?: Api_Contact;
+  public contactHolder?: ApiGen_Concepts_Contact;
   public depositReturnCount: number;
 
-  public constructor(baseDeposit: Api_SecurityDeposit) {
+  public constructor(baseDeposit: ApiGen_Concepts_SecurityDeposit) {
     this.id = baseDeposit.id || -1;
-    if (baseDeposit.depositType.id === 'OTHER') {
+    if (baseDeposit.depositType?.id === 'OTHER') {
```

Comment:
depositType is also a non-nullable field on PIMS_SECURITY_DEPOSIT. Why is this coming from the generated entity?

Reply:
because the codetype is an object the backend does not guarantee that it wont be null